course6/string_ariel/str.h

```diff
@@ -0,0 +1,193 @@
+#pragma once
+
+#include <iostream>
+#include <cstring>
+
+class Str
+{
+public:
+    Str()
+    {
+        start = mid = end = nullptr;
+    }
+
+    Str(const char* s)
+    {
+        int size = std::strlen(s);
+        start = new char[size];
+        end = start;
+
+        while (*s != '\0')
+        {
+            *end++ = *s++;
+        }
+
+        mid = end;
+    }
+
+    Str(const Str& str)
+    {
+        int size = str.Size();
+        start = new char[size];
+        end = start;
+
+        for (int i = 0; i < size; i++)
+        {
+            *end++ = str[i];
+        }
+
+        mid = end;
+    }
+
+    Str(Str&& str) :start(str.start), mid(str.mid), end(str.end)
+    {
+        str.start = str.mid = str.end = nullptr;
+    }
+
+    ~Str()
+    {
+        Dispose();
+    }
+
+    char operator[](int i) const
+    {
+        return *(start + i);
```

Comment:
start[i]

Reply:
Yeah, one translates to the other, I think I was stuck in pointer arithmetics at that point of the code